src/types/pysuper.rs

```diff
@@ -0,0 +1,61 @@
+use crate::ffi;
+use crate::type_object::PyTypeInfo;
+use crate::types::PyType;
+use crate::{PyAny, PyResult, Python};
+
+/// Represents a Python `super` object.
+///
+/// This type is immutable.
+#[repr(transparent)]
+pub struct PySuper(PyAny);
+
+pyobject_native_type_core!(PySuper, ffi::PySuper_Type);
+
+impl PySuper {
+    /// Constructs a new super object. More read about super object: [docs](https://docs.python.org/3/library/functions.html#super)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    ///
+    ///#[pyclass(subclass)]
+    /// struct BaseClass {
+    ///     val1: usize,
+    /// }
+    ///
+    /// #[pymethods]
+    /// impl BaseClass {
+    ///     #[new]
+    ///     fn new() -> Self {
+    ///         BaseClass { val1: 10 }
+    ///     }
+    ///
+    ///     pub fn method(&self) -> usize {
+    ///         self.val1
+    ///     }
+    /// }
+    ///
+    /// #[pyclass(extends=BaseClass)]
+    /// struct SubClass {}
+    ///
+    /// #[pymethods]
+    /// impl SubClass {
+    ///     #[new]
+    ///     fn new() -> (Self, BaseClass) {
+    ///         (SubClass {}, BaseClass::new())
+    ///     }
+    ///
+    ///     fn method<'a>(self_: PyRef<'_, Self>, py: Python<'a>) -> PyResult<&'a PyAny> {
+    ///         let any: Py<PyAny> = self_.into_py(py);
+    ///         let super_ = any.into_ref(py).py_super()?;
+    ///         super_.call_method("method", (), None)
+    ///     }
+    /// }
+    /// ```
+    pub fn new<'py>(py: Python<'py>, ty: &'py PyType, obj: &'py PyAny) -> PyResult<&'py PySuper> {
```

Comment:
Given that this takes gil-bound types `&'py PyType` and `&'py PyAny`, I think we can skip taking `py: Python<'py>`:

```suggestion
    pub fn new<'py>(ty: &'py PyType, obj: &'py PyAny) -> PyResult<&'py PySuper> {
        let py = ty.py();
```

Reply:
Thank you! Done